libsolidity/interface/CompilerStack.cpp

```diff
@@ -368,7 +368,10 @@ bool CompilerStack::analyze()
 		DeclarationTypeChecker declarationTypeChecker(m_errorReporter, m_evmVersion);
 		for (Source const* source: m_sourceOrder)
 			if (source->ast && !declarationTypeChecker.check(*source->ast))
+			{
+				m_hasError = true;
 				return false;
```

Comment:
Can you fix this in a uniform way, by either moving the inner part of the try/catch into a function that returns a bool and the caller of that function sets `m_hasError` appropriately? Then the `if (noErrors)` can be replaced by a return.
 -

Reply:
I made a few commits to simplify review, should be squashed afterwards.

Minor behavior changes are introduced in daba02b, they cause a few tests to fail (including command line tests). I updated the expectations, please double check if it is OK.